canonical/ubuntu-pro-client
# Purge option for RepoEntitlements

This is intended to land after #2744 - it contains the commits from there and will be rebased on whatever lands on that side.

still missing: integration tests

## Why is this needed?
This PR solves all of our problems because it implements the purge option when disabling services involving apt repositories. This removes/downgrades all the packages from that specific source.

## Test Steps
Integration tests still to come - but generally run disable for all services with `--purge` and see what happens :D

## Checklist

## Does this PR require extra reviews?
